components/post_view/channel_intro_message/add_members_button.tsx

```diff
@@ -35,25 +35,37 @@ export interface AddMembersButtonProps {
 }
 
 const AddMembersButton: React.FC<AddMembersButtonProps> = ({totalUsers, usersLimit, channel, setHeader, createBoard}: AddMembersButtonProps) => {
+    const currentTeamId = useSelector(getCurrentTeamId);
+
     if (!totalUsers) {
         return (<LoadingSpinner/>);
     }
 
     const isPrivate = channel.type === Constants.PRIVATE_CHANNEL;
     const inviteUsers = totalUsers < usersLimit;
-    const currentTeamId = useSelector(getCurrentTeamId);
 
     return (
         <TeamPermissionGate
             teamId={currentTeamId}
             permissions={[Permissions.ADD_USER_TO_TEAM, Permissions.INVITE_GUEST]}
         >
-            {inviteUsers && !isPrivate ? lessThanMaxFreeUsers(setHeader, createBoard) : moreThanMaxFreeUsers(channel, setHeader, createBoard)}
+            {inviteUsers && !isPrivate ? (
+                <LessThanMaxFreeUsers
+                    createBoard={createBoard}
+                    setHeader={setHeader}
+                />
+            ) : (
+                <MoreThanMaxFreeUsers
+                    channel={channel}
+                    createBoard={createBoard}
+                    setHeader={setHeader}
+                />
+            )}
         </TeamPermissionGate>
     );
 };
 
-const lessThanMaxFreeUsers = (setHeader: React.ReactNode, createBoard: React.ReactNode) => {
+const LessThanMaxFreeUsers = ({setHeader, createBoard}: {setHeader: React.ReactNode; createBoard: React.ReactNode}) => {
```

Comment:
This one was another weird one I couldn't figure out how to do without rewriting these entirely. Since they weren't technically functional components before, we weren't allowed to call `useIntl` in them.